Meminfo fails on WSL

On a machine running WSL (I used Arch but it shouldn't matter):

While the build passes, trying to run the `mem_info` function fails at runtime because WSL doesn't report `MemAvailable` (which is like several old Linux kernels.)

```
[hoverbear@Obsidian sys-info-rs]$ cargo test
warning: c/linux.c: In function ‘get_loadavg’:
warning: c/linux.c:62:12: warning: ‘avg’ is used uninitialized in this function [-Wuninitialized]
warning:      return avg;
warning:             ^~~
warning: c/linux.c: In function ‘get_mem_info’:
warning: c/linux.c:76:12: warning: ‘info’ is used uninitialized in this function [-Wuninitialized]
warning:      return info;
warning:             ^~~~
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running target/debug/deps/sys_info-0b3fbeb5431d68b8

running 10 tests
test test::test_cpu_num ... ok
test test::test_boottime ... ok
test test::test_cpu_speed ... ok
test test::test_disk_info ... ok
test test::test_loadavg ... ok
test test::test_os_release ... ok
test test::test_os_type ... ok
test test::test_proc_total ... ok
test test::test_mem_info ... FAILED
test test::test_hostname ... ok

failures:

---- test::test_mem_info stdout ----
thread 'test::test_mem_info' panicked at 'called `Result::unwrap()` on an `Err` value: Unknown', src/libcore/result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.


failures:
    test::test_mem_info

test result: FAILED. 9 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--lib'
[hoverbear@Obsidian sys-info-rs]$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

You can see the output from `/proc/meminfo`:

It's failing at:

https://github.com/FillZpp/sys-info-rs/blob/379f86da88ca58828001c8e5cece69a4b688f9db/lib.rs#L264

```
[hoverbear@Obsidian sys-info-rs]$ cat /proc/meminfo
MemTotal:       33462384 kB
MemFree:        22064676 kB
Buffers:           34032 kB
Cached:           188576 kB
SwapCached:            0 kB
Active:           167556 kB
Inactive:         157876 kB
Active(anon):     103104 kB
Inactive(anon):    17440 kB
Active(file):      64452 kB
Inactive(file):   140436 kB
Unevictable:           0 kB
Mlocked:               0 kB
SwapTotal:      60969468 kB
SwapFree:       60920476 kB
Dirty:                 0 kB
Writeback:             0 kB
AnonPages:        102824 kB
Mapped:            71404 kB
Shmem:             17720 kB
Slab:              13868 kB
SReclaimable:       6744 kB
SUnreclaim:         7124 kB
KernelStack:        2848 kB
PageTables:         2524 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:      515524 kB
Committed_AS:    3450064 kB
VmallocTotal:     122880 kB
VmallocUsed:       21296 kB
VmallocChunk:      66044 kB
HardwareCorrupted:     0 kB
AnonHugePages:      2048 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
DirectMap4k:       12280 kB
DirectMap4M:      897024 kB
```

You can see how this problem as fixed in another repo: https://github.com/polybar/polybar/issues/405 which lead to https://github.com/polybar/polybar/pull/410

--- a/lib.rs
+++ b/lib.rs
@@ -258,14 +258,25 @@ pub fn mem_info() -> Result<MemInfo, Error> {
                 meminfo_hashmap.insert(label, value);
             }
         }
+        let total = *meminfo_hashmap.get("MemTotal").ok_or(Error::Unknown)?;
+        let free = *meminfo_hashmap.get("MemFree").ok_or(Error::Unknown)?;
+        let buffers = *meminfo_hashmap.get("Buffers").ok_or(Error::Unknown)?;
+        let cached = *meminfo_hashmap.get("Cached").ok_or(Error::Unknown)?;
+        let avail = meminfo_hashmap.get("MemAvailable").map(|v| v.clone()).or_else(|| {
+            let sreclaimable = *meminfo_hashmap.get("SReclaimable")?;
+            let shmem = *meminfo_hashmap.get("Shmem")?;
+            Some(free + buffers + cached + sreclaimable - shmem)
+        }).ok_or(Error::Unknown)?;
+        let swap_total = *meminfo_hashmap.get("SwapTotal").ok_or(Error::Unknown)?;
+        let swap_free = *meminfo_hashmap.get("SwapFree").ok_or(Error::Unknown)?;
         Ok(MemInfo {
-            total: *meminfo_hashmap.get("MemTotal").ok_or(Error::Unknown)?,
-            free: *meminfo_hashmap.get("MemFree").ok_or(Error::Unknown)?,
-            avail: *meminfo_hashmap.get("MemAvailable").ok_or(Error::Unknown)?,
-            buffers: *meminfo_hashmap.get("Buffers").ok_or(Error::Unknown)?,
-            cached: *meminfo_hashmap.get("Cached").ok_or(Error::Unknown)?,
-            swap_total: *meminfo_hashmap.get("SwapTotal").ok_or(Error::Unknown)?,
-            swap_free: *meminfo_hashmap.get("SwapFree").ok_or(Error::Unknown)?,
+            total,
+            free,
+            avail,
+            buffers,
+            cached,
+            swap_total,
+            swap_free,
         })
     } else if cfg!(target_os = "macos") || cfg!(target_os = "windows") {
         Ok(unsafe { get_mem_info() })

